src/middlewared/middlewared/pytest/unit/utils/test_statx.py

```diff
@@ -0,0 +1,105 @@
+import os
+import pytest
+import stat
+
+from middlewared.plugins.filesystem_ import stat_x as sx
+
+
+BASIC_STAT_ATTRS = [
+    'MODE',
+    'UID',
+    'GID',
+    'ATIME',
+    'MTIME',
+    'CTIME',
+    'DEV',
+    'INO',
+    'SIZE',
+    'BLOCKS',
+    'BLKSIZE',
+    'NLINK',
+]
+
+def timespec_convert(timespec):
+    return timespec.tv_sec + timespec.tv_nsec / 1000000000
```

Comment:
Hmmm.  What are you trying to do with this calculation?  **round** to nearest second, **floor** to the second?

Either way, presumably `timespec.tv_nsec / 1000000000` will always be < 1

AFAICS `os.stat` yields a whole int.

Reply:
```
>>> os.stat('/tmp')
os.stat_result(st_mode=17407, st_ino=1, st_dev=44, st_nlink=9, st_uid=0, st_gid=0, st_size=180, st_atime=1715354418, st_mtime=1715357642, st_ctime=1715357642)
>>> os.stat('/tmp').st_mtime
1715357642.7296498
>>>
```
Weird, but OK